src/solvers/strings/string_refinement.cpp

```diff
@@ -1628,34 +1628,20 @@ compute_inverse_function(const exprt &qvar, const exprt &val, const exprt &f)
   return sum_over_map(elems, f.type(), neg);
 }
 
-class find_qvar_visitort : public const_expr_visitort
-{
-private:
-  const exprt &qvar_;
-
-public:
-  bool found;
-
-  explicit find_qvar_visitort(const exprt &qvar) : qvar_(qvar), found(false)
-  {
-  }
-
-  void operator()(const exprt &expr) override
-  {
-    if(expr == qvar_)
-      found = true;
-  }
-};
-
 /// look for the symbol and return true if it is found
 /// \param index: an index expression
 /// \param qvar: a symbol expression
 /// \return a Boolean
 static bool find_qvar(const exprt &index, const symbol_exprt &qvar)
 {
-  find_qvar_visitort v2(qvar);
-  index.visit(v2);
-  return v2.found;
+  bool found = false;
+
+  index.visit([&found, &qvar](const exprt &e) {
+    if(e == qvar)
+      found = true;
```

Comment:
I would rather write `std::find(e.depth_begin(), e.depth_end(), qvar)`, but this version is still better than the previous one.